zipkin-lens/src/components/DiscoverPage/DiscoverPageContent.test.jsx

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright 2015-2020 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+/* eslint-disable react/prop-types */
+
+import { act, renderHook } from '@testing-library/react-hooks';
+import { createMemoryHistory } from 'history';
+import moment from 'moment';
+import React from 'react';
+import { Router } from 'react-router-dom';
+
+import DiscoverPageContent, {
+  buildApiQuery,
+  parseDuration,
+  useQueryParams,
+} from './DiscoverPageContent';
+import render from '../../test/util/render-with-default-settings';
+
+describe('useQueryParams', () => {
+  it('should extract criteria from query string', () => {
+    const history = createMemoryHistory();
+    const wrapper = ({ children }) => {
+      return <Router history={history}>{children}</Router>;
+    };
+
+    history.push({
+      pathname: '/zipkin/',
+      // serviceName: serviceA
+      // spanName: spanB
+      // remoteServiceName: remoteServiceNameC
+      // minDuration: 10us
+      // maxDuration: 100ms
+      // annotationQuery:
+      //   key1: value1
+      //   key2
+      //   key3: value3
+      search:
+        '?serviceName=serviceA&spanName=spanB&remoteServiceName=remoteServiceNameC&minDuration=10us&maxDuration=100ms&annotationQuery=key1%3Dvalue1+and+key2+and+key3%3Dvalue3&limit=10',
+    });
+
+    const { result } = renderHook(() => useQueryParams(['key3']), { wrapper });
+    expect(result.current.criteria).toEqual([
+      { key: 'serviceName', value: 'serviceA' },
+      { key: 'spanName', value: 'spanB' },
+      { key: 'remoteServiceName', value: 'remoteServiceNameC' },
+      { key: 'minDuration', value: '10us' },
+      { key: 'maxDuration', value: '100ms' },
+      // AnnotationQuery
+      { key: 'key3', value: 'value3' },
+      { key: 'tagQuery', value: 'key1=value1 and key2' },
```

Comment:
I know tagQuery doesn't match the old rest api annotationQuery, but chatted with @tacigar and I think these days almost never someone would put an "annotation" in there.

anyone object to having the UI say tagQuery not annotationQuery? (the rest api is the same regardless as not important enough to change) @openzipkin/core